File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
       lastdigit = len(nums)-1
       start =0
       while(start <= lastdigit ):
           if(nums[start]== val and nums[lastdigit] != val):
               nums[start] = nums[lastdigit]
               lastdigit -= 1
           if(nums[lastdigit] == val):
               lastdigit -= 1
           if(nums[start] != val):
               start += 1

       return start





        

File: 0027-remove-element/0027-remove-element.c
int removeElement(int* nums, int numsSize, int val) {
        int j = numsSize-1;
        int i=0;
        while(i<=j){
            if(nums[i] == val && nums[j] != val){
                nums[i] = nums[j];
                j--;
            }
            if(nums[j] == val){
                j--;
            }
            if(nums[i] != val){
                i++;
            }
        }
        return i;
}